web/web.go

```diff
@@ -111,6 +112,8 @@ func NewWebService(o *WebServiceOptions) *WebService {
 		router.Post("/-/quit", ws.quitHandler)
 	}
 
+	router.Get("/federate", instr("federate", o.Federation))
```

Comment:
Yes, was thinking the same - at least about versioning it.
It would certainly be a special endpoint as it doesn't talk the same way as the other endpoints.

In general I feel changes to federation and the API will not synchronize well.